### ISSUE TITLE
Enable --no-capture with --verbose.

### ISSUE THREAD

**Issue by EvanCarroll** [rust-lang/cargo#10777]
### Problem

This interface is unintuivie 100k people have viewed this question on StackOverflow,

https://stackoverflow.com/q/25106554/124486

It has 462 upvotes. The top answer has 533 up votes. The expectation of any user is going to be to display _more_ information you enable `--verbose`. It seems highly likely that the information that they're going to want to see is either all of it, or specifically the information that's not shown.

### Proposed Solution

A suggestion to fix this, is for verbose to take multiple options with the default being `all`.

* `--verbose=all` display everything you can.
* `--verbose=subset` for whatever the current `--verbose` does.

### Notes

And, it'd be nice if the [current behavior of `--verbose` was even documented](https://stackoverflow.com/q/72693927/124486).

**Comment by EvanCarroll** [comment]
If the proposed solution sounds acceptable, I'll volunteer to write the patch.

**Comment by weihanglo** [comment]
FWIW, the behaviour is indeed [mentioned in `cargo help test`](https://doc.rust-lang.org/cargo/commands/cargo-test.html#display-options), which is right above `--verbose` flag.

**Comment by Muscraft** [comment]
The [`rustc book`](https://doc.rust-lang.org/rustc/) also mentions [`--nocapture`](https://doc.rust-lang.org/rustc/tests/index.html#--nocapture) and [`--show-output`](https://doc.rust-lang.org/rustc/tests/index.html#--show-output) under [Output options](https://doc.rust-lang.org/rustc/tests/index.html#output-options) in the [Tests page](https://doc.rust-lang.org/rustc/tests/index.html)

**Comment by daweth** [comment]
This feature needs to be added, either as a command flag to `cargo test` or some other way, not being able to print natively makes it extremely difficult to debug tests

**Comment by tbu-** [comment]
> This feature needs to be added, either as a command flag to `cargo test` or some other way, not being able to print natively makes it extremely difficult to debug tests

You can already do this, pass `-- --nocapture` or `-- --show-output` to `cargo test`.

**Comment by tbu-** [comment]
> Point was never that it couldn't be done.

I was trying to respond to the post I quoted, which *sounded* to me like implying that it couldn't be done. I was trying to help that person. Is that a bad thing? ^^

**Comment by EvanCarroll** [comment]
Not at all, just trying to keep it on topic while you're helping him. ;) We need to make it such that the intuitive thing -- adding `--verbose` -- does what people expect. This is only a UX issue.

**Comment by SJonesy** [comment]
Stopping by to say that today I didn't understand why `cargo test --verbose` wouldn't tell me which assert in my test was failing. I googled it, found the referenced SO answer, and now I'm here.

**Comment by rimutaka** [comment]
**What if we add a prompt to use `--nocapture` if a test fails?**
It is already done in https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_test.rs for abnormal test completion:
```rust
    if !is_simple && executed && harness && !nocapture {
        drop(ws.gctx().shell().note(
            "test exited abnormally; to see the full output pass --nocapture to the harness.",
        ));
    }
```
@EvanCarroll , @ehuss , @weihanglo , I'm happy to look into it if you approve.

**Comment by epage** [comment]
> What if we add a prompt to use --nocapture if a test fails?

If a test fails, libtest already outputs what was captured, so I'm not sure what this would buy us.

The reason for that message is the test binary might have crashed before it had a chance to tell the user what failed (or what test it was on when it crashed)
